app/shared/Shared/Models/RequestOverrides.cs

```diff
@@ -2,11 +2,17 @@
 
 namespace Shared.Models;
 
+public enum RetrievalMode
```

Comment:
This should really be in its own file, and since they're public APIs we should provide triple slash comments, for example:

```cs
/// <summary>Represents the various retrieval mode options.</summary>
public enum RetrievalMode
{
    /// <summary>
    /// Represents a text retrieval mode, that will (some explanation of what it does)...
    /// </summary>
    Text = 0,
    // omitted for brevity...
}
```

We need to detail their intended usage.